data/src/main/java/com/nexters/knownknowns/data/repositoryimpl/UserRepositoryImpl.kt

```diff
@@ -1,34 +1,42 @@
 package com.nexters.knownknowns.data.repositoryimpl
 
+import com.nexters.knownknowns.data.datasource.AuthDataSource
 import com.nexters.knownknowns.data.datasource.UserDataSource
 import com.nexters.knownknowns.data.local.user.ClickState
-import com.nexters.knownknowns.data.local.user.UserDataStore
 import com.nexters.knownknowns.data.model.request.UserInfoRequest
 import com.nexters.knownknowns.data.repository.UserRepository
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flow
 import org.koin.core.annotation.Single
 
 @Single
 internal class UserRepositoryImpl(
-    private val userDataStore: UserDataStore,
-    private val userDataSource: UserDataSource
+    private val authDataSource: AuthDataSource,
+    private val userDataSource: UserDataSource,
 ) : UserRepository {
-    override val clickStateFlow: Flow<ClickState>
-        get() = userDataStore.clickStateFlow
+    override val clickStateFlow: Flow<ClickState> = userDataSource.clickStateFlow
 
     override suspend fun incrementClickCount() {
-        userDataStore.incrementClickCount()
+        userDataSource.incrementClickCount()
     }
 
     override suspend fun resetClickState() {
-        userDataStore.resetClickState()
+        userDataSource.resetClickState()
     }
 
     override suspend fun recordBottomSheetShown() {
-        userDataStore.recordBottomSheetShown()
+        userDataSource.recordBottomSheetShown()
     }
 
-    override suspend fun putUserInfo(request: UserInfoRequest): Result<Unit> = runCatching {
-            userDataSource.putUserInfo(request)
+    override fun putUserInfo(request: UserInfoRequest): Flow<Unit> = flow {
+        val userId = authDataSource.getUserId().first()
+
+        emit(
+            userDataSource.putUserInfo(
+                userId = userId ?: 0L,
+                request = request
+            )
+        )
```

Comment:
emit은 flow가 아닌 곳에서 flow를 만드는 거라, authDataSource.getUserId() 자체가 flow라면 map으로도 할 수 있어!
```suggestion
    override fun putUserInfo(request: UserInfoRequest): Flow<Unit> = authDataSource.getUserId().map {  
            userDataSource.putUserInfo(
                userId = userId ?: 0L,
                request = request
            )
    }
```

Reply:
오! flow 아직 더 공부해야겠다ㅜㅜ 수정했습니다!